lib/runner/extract-runnable-items.js

```diff
@@ -61,6 +61,49 @@ var sdk = require('postman-collection'),
         return matched;
     },
 
+    /**
+     * Finds items based on multiple ids or names provided.
+     *
+     * @param {ItemGroup} itemGroup - Composite list of Item or ItemGroup.
+     * @param {Object} entrypointSubset - Entry-points reference passed across multiple recursive calls.
+     * @param {Boolean} recursive - Flag used to decide whether to include nested items or not.
+     * @param {Array<String>} _accumulatedItems - Found Items or ItemGroups.
+     * @returns {Object}
+     */
+    findItemsOrGroups = function (itemGroup, entrypointSubset, recursive, _accumulatedItems) {
+        if (!itemGroup || !itemGroup.items) { return; }
+
+        !_accumulatedItems && (_accumulatedItems = []);
+
+        var idOrName;
+
+        itemGroup.items.each(function (item) {
+            // bail out if all entry-points are found.
+            if (!Object.keys(entrypointSubset).length) { return false; }
+
+            // lookup for item.id in entrypointSubset.
+            idOrName = entrypointSubset[item.id] && item.id;
+
+            // if item.id not found, lookup by item.name in entrypointSubset.
+            !idOrName && (idOrName = entrypointSubset[item.name] && item.name);
+
+            if (idOrName) {
+                // only push items which are not previously got tracked from its parent entry-point.
+                recursive && _accumulatedItems.push(item);
+
+                // delete looked-up entry-point.
+                delete entrypointSubset[idOrName];
+
+                // recursive call to find nested entry-points. To make sure all provided entry-points got tracked.
+                // set recursive flag to `false` for children, since current item is included already.
+                return findItemsOrGroups(item, entrypointSubset, false, _accumulatedItems);
+            }
+
+            findItemsOrGroups(item, entrypointSubset, recursive, _accumulatedItems);
```

Comment:
```
// recursive call to find nested entry-points. To make sure all provided entry-points got tracked.
// set recursive flag to `false` for children, since current item is included already.
return findItemsOrGroups(item, entrypointSubset, !idOrName, _accumulatedItems)
```

With this, we do not need multiple return paths.